### PR TITLE
Created API for TensorGraph optimizers

### LINKED ISSUE #700: TensorGraph optimizers
I've been looking into using learning rate decay for RL, and I realized the method in #680 won't work for it.  That method relies on first creating the TensorGraph so you can get a reference to its global step variable, then using that when creating the optimizer.  But A3C requires creating the optimizer first and passing it to the constructor.  I could probably restructure it to allow that API to be changed.  But I've started wondering whether we really should just create our own set of optimizer classes for TensorGraph.  Instead of writing

```Python
optimizer = TFWrapper(tf.train.AdamOptimizer, learning_rate=0.0001)
```

you would write

```Python
optimizer = optimizers.Adam(learning_rate=0.0001)
```

Or if you wanted learning rate decay,

```Python
learning_rate = optimizers.polynomial_decay(initial=0.001, final=0.0001, steps=500000)
optimizer = optimizers.Adam(learning_rate=learning_rate)
```

It would get rid of the ugliness with `TFWrapper`, and simplify the API since users wouldn't have to deal with the `global_step` tensor.  It also would mean one less bit of explicit TensorFlow code in the API, which will be helpful if we ever decide to switch from TensorFlow to PyTorch or something else for the back end.

### PR DESCRIPTION
**Pull request by peastman**
Fixes #700.  This simplifies the API and allows learning rate decay to be used with RL.  Initially I've only included two optimizers (Adam and gradient descent), since those are the only ones ever used in the DeepChem code or the examples.  Likewise, I've only included two learning rate schedules (exponential and polynomial).  I'm not sure if there's much benefit to having more choices than that.  Anyway, it's easy to add more of either if we want them.

### CODE REVIEW

File: contrib/rl/tictactoe.py [hunk 1]
```diff
@@ -66,8 +67,7 @@ def eval_tic_tac_toe(value_weight,
         entropy_weight=0.01,
         value_weight=value_weight,
         model_dir=model_dir,
-        optimizer=dc.models.tensorgraph.TFWrapper(
-            tf.train.AdamOptimizer, learning_rate=0.001))
+        optimizer=Adam(learning_rate=0.001))
```
Comment (rbharath) [inline contrib/rl/tictactoe.py]: This new API looks beautiful!
